### PR TITLE
Control recursive pulls

### PR DESCRIPTION
**Pull request by richfitz**
This PR is the next bit of yak shaving; introduces a config option to require that packet pulls are recursive, and has support for changing that option. We'll use this to tidy up how locations provide packets.

~Merge after #8, as this PR includes those commits; see https://github.com/mrc-ide/outpack/compare/mrc-3030...mrc-3093 for the diff~

### CODE REVIEW

File: R/config.R [hunk 1]
```diff
@@ -1,4 +1,79 @@
-config_new <- function(path_archive, use_file_store) {
+##' Set configuration options. Not all can currently be set; this will
+##' be expanded over time. See Details.
+##'
+##' Options are set in the order that they are provided.  Currently,
+##' if setting one option fails, no further options will be processed
+##' but previous ones will not be (do not rely on this behaviour, it
```
Comment (r-ash) [inline R/config.R]:
```suggestion
##' but previous ones will be (do not rely on this behaviour, it
```
?

File: R/config.R [hunk 2]
```diff
@@ -1,4 +1,79 @@
...
+##'   named options here (e.g., `list(core.require_pull_recursive =
+##'   TRUE)`).  This interface is typically easier to program against.
+##'
+##' @inheritParams outpack_location_list
+##'
+##' @return Nothing
+##' @export
+outpack_config_set <- function(..., options = list(...), root = NULL) {
+  root <- outpack_root_locate(root)
+  if (!missing(options) && ...length() > 0) {
```
Comment (r-ash) [inline R/config.R:30]: I don't think I've seen the `...length()` syntax before, looks weird!

Comment (r-ash) [reply R/config.R:30]: Oh, that's just a function name :smile:

**Review by r-ash — APPROVED** [review]
One tiny typo but otherwise this looks great!

File: R/config.R (hunk 2 above)
Comment (richfitz) [reply R/config.R:30]: yeah, pretty odd! But it does the job here. I could have done

```
  if (!missing(options) && length(list(...)) > 0) {
```

but that function is the one designed for this job so went with the weird

**Review by hillalex — APPROVED** [review]
Makes sense to me!

**Pull request merged**